### ISSUE TITLE
NVDA fails to delete temporary files with .delete extension

### ISSUE THREAD

**Issue by paulber19** [nvaccess/nvda#17562]
After a faulty installation of an add-on, a temporary file with the .delete extension remains in the "addons" folder and NVDA reports an error in the log on each restart.
ERROR - addonHandler.removeFailedDeletion (17:59:45.736) - MainThread (3760):
Failed to delete path E:\documents\Paulo\temp\nvda 2024.4\userConfig\addons\tmpneiy66px.delete, try removing manually
### Actual behavior:

- choose an add-on in the addons folder
- open a file of this add-on in text editor
- update this add-on and restart NVDA
- nvda reports an error loading the add-on
- close the text editor and restart NVDA
- the add-on is correctly installed but an error is reported in the log.
### Expected behavior:

Since it is possible to manually delete this fileNVDA should be able to do that too à when it starts.
#### NVDA installed/portable/running from source:
NVDA portable
#### NVDA version:
NVDA 2024.4.1
#### Windows version:
Windows 10 22H2 (10.0.19045) workstation AMD64
#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
An add-on is necessary.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
No tested.

**Comment by LeonarddeR** [comment]
I can reproduce this.
